src/testhost.x86/TestHostTraceListener.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+namespace Microsoft.VisualStudio.TestPlatform.TestHost
+{
+#if NETCOREAPP
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using System;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Reflection;
+
+    internal class TestHostTraceListener : TraceListener
+    {
+        public static void Setup()
+        {
+            EqtTrace.Info("Setting up debug trace listener.");
+            EqtTrace.Verbose("TestPlatformTraceListener.Setup: Removing listener {0}.", Trace.Listeners[0]);
+            Trace.Listeners[0] = new TestHostTraceListener();
```

Comment:
It may be better to call `Trace.Listeners.Clear()` first. 

However, I am not sure if this can interfere with TraceListeners that the user wants to register. I wonder if we should provide a runsetting or something else if user wants to opt-out of the default handling in TP and provide their own TraceListener...

Reply:
I ended up replacing all DefaultTraceListeners in that collection, typically there would be only one. If user manages to register their own before us, it will remain there.

Reply:
Ah I didn't realize DefaultTraceListeners was a publicly accessible type.